### PR TITLE
Secure Key feature and minor adjustments.

### PR DESCRIPTION
**Pull request by JesusLuvsYooh**
- Added secure key feature
(get list  is now post with key, to help protect possible sensitive data (user ids, model/platform etc?), functions reply with 400's if undefined or keys no match)

- Fixed add server player amount default 0
( new servers if client is hosting would be 1 already, if you have 2 split screen players on same device, it would be 2, basically player amount could be anything depending on your setup, and is nice to set from very beginning.)

- Added IP into update section
(Maqsooms edit)

### CODE REVIEW

File: listServer.js [hunk 1]
```diff
@@ -72,16 +76,27 @@ function denyRequest (req, res) {
 // --- Functions --- //
 // API: Returns JSON list of servers.
 function apiGetServerList(req, res) {
-	if(req.method != "GET") {
+	if(req.method != "POST") {
```
Comment (SoftwareGuy) [inline listServer.js:79]: These chunks are probably not needed anymore but I'll clean them up

File: listServer.js [hunk 2]
```diff
@@ -72,16 +76,27 @@ function denyRequest (req, res) {
...
+		//Shows if keys match for those getting list server details.
+		console.log(`[WARN] Keys match ${req.body.serverKey} ${secureLSKey}`);
+	}
 
 	// A client wants the server list. Compile it and send out via JSON.
 	var serverList = [];
 	
 	for (var i = 0, len = knownServers.length; i < len; i++) {
-		serverList.push({ 'ip': knownServers[i].ip, 'name': knownServers[i].name, 'port': parseInt(knownServers[i].port), 'players': parseInt(knownServers[i].players) })
+		serverList.push({ 'ip': knownServers[i].ip, 'name': knownServers[i].name, 'port': parseInt(knownServers[i].port), 'players': parseInt(knownServers[i].players), 'serverKey': knownServers[i].serverKey })
```
Comment (SoftwareGuy) [inline listServer.js]: Why are we sending the serverKey to the client? Is this going to pose as a security risk?

**Review by SoftwareGuy — CHANGES_REQUESTED** [review]
Looks good. Just need a little clarification on one thing, then I'll push the PR into master. 👍

File: listServer.js (hunk 1 above)
Comment (JesusLuvsYooh) [reply listServer.js:79]: Okay :)

**Pull request merged**